### DISCUSSION TITLE
Memory leak of ItemsControl

### DISCUSSION THREAD

**Discussion by LionelXN** [unoplatform/uno#11290] [Q&A] [answered]
I created a Demo gtk project for Uno Platform. When I add UI elements to ItemsControl, the memory will continue to increase, but when I remove it, it will not decrease until I call GC.Collect, but it will not be completely released. What should I do?

**Comment by FrozDark** [comment, marked as answer]
Fix has been merged recently https://github.com/unoplatform/uno/pull/11247

What version of Uno did you use when it's happened?

**Reply by LionelXN** [reply]
hi，i using Uno of  4.5.14，which version fixes this problem?

**Reply by jeromelaban** [reply]
There is no available build that contains the mentioned PR yet, it should be available during the next few hours as 4.8.0-dev.309 on nuget.

**Reply by LionelXN** [reply]
I used 4.8.0-dev.318 and found that it did not take effect, and I did not see the corresponding code for this submission
![image](https://user-images.githubusercontent.com/18021306/218676992-37ad7377-a8b1-4167-b422-12cc63ef5f98.png)

![image](https://user-images.githubusercontent.com/18021306/218676906-90bebfa9-e939-40b0-8357-1a638a1e2a87.png)

**Reply by LionelXN** [reply]
So,which version of Uno can  fix this problem on Nuget?

**Reply by jeromelaban** [reply]
The code you're seeing is the right one and contains the changes from #11247. If you have another leak, you may be hitting a scenario that we have not yet encountered. Also know that Uno uses `FrameworkTemplatePooling` which keeps some UI elements for a little while for performance reasons.

I did not see a repro in your original question. If you have one, make sure to have one and we'll track it in an issue.

**Reply by LionelXN** [reply]
Is there any way to control the UI elements that are retained; Because I kept adding and deleting ItemsSource, I observed that UI elements kept accumulating;

**Reply by LionelXN** [reply]
There is no way to free UI element resources autonomously

**Reply by LionelXN** [reply]
1.When is the UI retained by the FrameworkTemplatePooling operation Disposable?
2.If Disposable, what method is called?

**Reply by jeromelaban** [reply]
You can disable the pooling by using `FrameworkTemplatePooling.IsPoolingEnabled` and setting it to false.

As for disposing objects, no, none of the objects are disposable, object are automatically reclaimed by the GC when not referenced anymore.

If you can provide a repro for your issue, it will be easier to diagnose. Otherwise, use a memory profiling tool to determine what is keeping memory allocated.

**Reply by LionelXN** [reply]
Thanks for your answer，it helps a lot to improve memory leak when using FrameworkTemplatePool.IsPoolingEnabled = false;
but,what are the hidden dangers of using it this way?I didn't feel it during the trial?

**Reply by FrozDark** [reply]
Low perfomance on rendering views cause it have to create new ones from scratch

**Reply by jeromelaban** [reply]
Indeed, `FrameworkTemplatePool.IsPoolingEnabled = false;` will disable reuse and will degrade runtime performance as create new views is costly.

As disabling pooling reduces the memory increase you are seeing, this means that the pooling automatic cleanup will work properly and will decrease when the app is not used, or when the pooled templates are not used anymore by the app.
